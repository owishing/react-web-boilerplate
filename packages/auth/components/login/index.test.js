import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Login } from '.';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));

function shallowRenderWithProps(props = {}) {
  return shallow(<Login {...props} />);
}

test('Render right Login.', () => {
  const component = shallowRenderWithProps();
  expect(toJson(component)).toMatchSnapshot();
});
