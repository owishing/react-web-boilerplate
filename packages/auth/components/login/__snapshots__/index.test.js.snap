// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render right Login. 1`] = `
<div
  className="container"
>
  <div
    className="form"
  >
    <div
      className="title"
    >
      AppInsights
    </div>
    <WithStyles(ForwardRef(TextField))
      autoFocus={true}
      className="email"
      fullWidth={true}
      id="email"
      margin="normal"
      onChange={[Function]}
      placeholder="Email"
      size="small"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="password"
      fullWidth={true}
      id="password"
      margin="normal"
      onChange={[Function]}
      placeholder="Password"
      size="small"
      type="password"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(Button))
      className="submit"
      color="primary"
      disabled={false}
      fullWidth={true}
      onClick={[Function]}
      type="submit"
      variant="contained"
    >
      LOG IN
    </WithStyles(ForwardRef(Button))>
    <div
      className="tools"
    >
      <WithStyles(ForwardRef(FormControlLabel))
        className="remember-me"
        control={
          <ForwardRef(WithStyles)
            color="primary"
            value="remember"
          />
        }
        label="Remember me"
      />
      <WithStyles(ForwardRef(Link))
        className="forgot"
      >
        Forgot password?
      </WithStyles(ForwardRef(Link))>
    </div>
  </div>
</div>
`;
